js/compiler_tests.ts

```diff
@@ -0,0 +1,83 @@
+import { test, assert } from "./testing/testing.ts";
+import { DenoCompiler, Os, Ts } from "compiler";
+// import { test, assert } from "./testing/testing";
+// import { DenoCompiler, Os, Ts } from "./compiler";
+import * as ts from "typescript";
+
+interface ModuleInfo {
+    moduleName: string | null;
+    filename: string | null;
+    sourceCode: string | null;
+    outputCode: string | null;
+}
+
+const osMock: Os = {
+    codeCache(
+        filename: string,
+        sourceCode: string,
+        outputCode: string
+    ): void {},
+    codeFetch(moduleSpecifier: string, containingFile: string): ModuleInfo {
+        return {
```

Comment:
Add a counter to the mocked methods so we can check how many times they were called.